orc8r/cloud/go/tools/combine_swagger/spec/combine.go

```diff
@@ -17,189 +17,53 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
-	"regexp"
-	"sort"
 	"strings"
-
-	"magma/orc8r/cloud/go/tools/swaggergen/generate"
-
-	"github.com/hashicorp/go-multierror"
-	"github.com/pkg/errors"
-	"gopkg.in/yaml.v2"
 )
 
-// Load specs from file to Swagger structs.
-func Load(inDir, commonFilepath string) ([]generate.SwaggerSpec, generate.SwaggerSpec, error) {
+// Load specs from file to YAML string
+func Load(inDir, commonFilepath string) ([]string, string, error) {
 	specs, err := loadSpecsFromInputDir(inDir)
 	if err != nil {
-		return nil, generate.SwaggerSpec{}, err
+		return nil, "", err
 	}
 
-	commonSpec, err := loadCommonSpec(commonFilepath)
+	commonSpec, err := readFile(commonFilepath)
 	if err != nil {
-		return nil, generate.SwaggerSpec{}, err
+		return nil, "", err
 	}
 
 	return specs, commonSpec, nil
 }
 
-// Combine multiple Swagger specs, giving precedence to the "common" spec.
-// Returned "error" contains warnings for any overwritten fields.
-//
-// This custom-built functionality mirrors the "official" implementation.
-// See: https://github.com/go-openapi/analysis/blob/master/mixin.go
-func Combine(common generate.SwaggerSpec, specs []generate.SwaggerSpec) (generate.SwaggerSpec, error) {
-	var tags [][]generate.TagDefinition
-	var paths []map[string]interface{}
-	var responses []map[string]interface{}
-	var parameters []map[string]interface{}
-	var definitions []map[string]interface{}
-
-	for _, s := range specs {
-		tags = append(tags, s.Tags)
-		paths = append(paths, s.Paths)
-		responses = append(responses, s.Responses)
-		parameters = append(parameters, s.Parameters)
-		definitions = append(definitions, s.Definitions)
-	}
-
-	errs := &multierror.Error{}
-
-	out := generate.SwaggerSpec{
-		Swagger:  common.Swagger,
-		Info:     common.Info,
-		BasePath: common.BasePath,
-		Consumes: common.Consumes,
-		Produces: common.Produces,
-		Schemes:  common.Schemes,
-
-		Tags:        combineTags(common.Tags, tags, errs),
-		Paths:       combineSubSpec(common.Paths, paths, "paths", errs),
-		Responses:   combineSubSpec(common.Responses, responses, "responses", errs),
-		Parameters:  combineSubSpec(common.Parameters, parameters, "parameters", errs),
-		Definitions: combineSubSpec(common.Definitions, definitions, "definitions", errs),
+// loadSpecsFromInputDir loads all input Swagger files' contents
+// to string.
+func loadSpecsFromInputDir(inDir string) ([]string, error) {
+	filepaths := getFilepaths(inDir)
+	contents, err := readFiles(filepaths)
+	if err != nil {
+		return nil, err
 	}
 
-	return out, errs.ErrorOrNil()
+	return contents, nil
 }
 
 // Write spec to filepath.
-func Write(spec generate.SwaggerSpec, filepath string) error {
-	strSpec, err := marshalToYAML(spec)
+func Write(outSpec string, filepath string) error {
```

Comment:
In general, especially for shorter files, keep public function definitions grouped near the top of the file (and before / not interwoven with private fn definitions)

Reply:
Got it. Will keep that in mind